### PR TITLE
fix(charts-angular): update build toolchain

### PR DESCRIPTION
**Pull request by cal-smith**
- switches the build to use the angular cli
- updates the tsconfig to a known working state (based on the default angular cli output)
- syncs all the build package versions
- fixes build for ng6

### CODE REVIEW

**Comment by cal-smith** [comment]
Just doing some local validation:

- [x] ng6 works
- [ ] ng7 works
- [ ] ng8 works

**Comment by cal-smith** [comment]
All good from a compatibility perspective ... now just sorting out the travis issues

**Comment by cal-smith** [comment]
Travis sorted, seems like netlify was using an outdated node version, so I bumped that - the cache will need to be cleared though before it will build correctly (mainly for node-sass ... we should swap out to sass to avoid that in the future)

File: packages/angular/tsconfig.json [hunk 1]
```diff
@@ -2,19 +2,35 @@
...
+    "declaration": true,
+	"moduleResolution": "node",
+	"module": "es2015",
     "emitDecoratorMetadata": true,
-    "experimentalDecorators": true,
-    "target": "es5",
+	"experimentalDecorators": true,
+	"downlevelIteration": true,
+	"importHelpers": true,
+    "target": "es2015",
```
Comment (theiliad) [inline packages/angular/tsconfig.json]: I think there are some mixed indentation types in this file

File: .travis/before_install.sh [hunk 2]
```diff
@@ -2,4 +2,4 @@
 
 set -e
 
-yarn global add lerna@3.13.4
+# yarn global add lerna@3.13.4
```
Comment (theiliad) [inline .travis/before_install.sh]: we needed this in the past? Is that not the case anymore?

File: .travis.yml [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
+  - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.21.1
   - export PATH="$HOME/.yarn/bin:$PATH"
   - .travis/before_install.sh
   - "google-chrome-stable --headless --disable-gpu --remote-debugging-port=9222 http://localhost &"
 install: yarn --offline
 script:
-  - lerna run lint
-  - lerna run test
+  - yarn run lint
+  - yarn run test
```
Comment (theiliad) [inline .travis.yml]: would yarn workspaces pick that up and run in all packages?

File: .nvmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+11
```
Comment (theiliad) [inline .nvmrc]: does this tell travis what node version to use?

File: .travis/before_install.sh (hunk 2 above)
Comment (cal-smith) [reply .travis/before_install.sh]: I swapped the travis commands to just use the root package.json scripts 👍

File: .nvmrc (hunk 4 above)
Comment (cal-smith) [reply .nvmrc]: It's for netlify ... as far as I can see it's the only way to specify the node version there ... travis uses the `.travis.yml`

File: .travis/before_install.sh (hunk 2 above)
Comment (theiliad) [reply .travis/before_install.sh]: So let's remove that line completely then?

File: .travis.yml (hunk 3 above)
Comment (cal-smith) [reply .travis.yml]: That runs the scripts from the root package.json, which in turn run `lerna run x` ... so yeah ... this just makes the environment more consistent

File: packages/angular/tsconfig.json (hunk 1 above)
Comment (cal-smith) [reply packages/angular/tsconfig.json]: 🤦‍♂

File: .travis/before_install.sh (hunk 2 above)
Comment (cal-smith) [reply .travis/before_install.sh]: I can remove the commented out command

Comment (cal-smith) [reply .travis/before_install.sh]: Sorry, submitted after you commented ... I'll do that 👍

**Comment by theiliad** [comment]
closing & openning the PR only causes travis to rebuild. could you confirm locally that things work?  I don't think this is an issue

**Comment by theiliad** [comment]
Ran the `demo:build` script locally, works fine

**Comment by cal-smith** [comment]
> closing & openning the PR only causes travis to rebuild. could you confirm locally that things work? I don't think this is an issue

Yeah it's just netlify exploding since the node version changed ... Its trying to run node 8 node-sass on node 11

**Pull request merged**